docs/release-management.md

```diff
@@ -0,0 +1,139 @@
+        * and this other one
+      - upgrades: >  # optional
+        Upgrade notes important to anybody upgrading from a previous version
+      - deprecations: > # optional
+        Features or settings that have been deprecated and need to be updated
+        by the administrator.
+
+    components:
+      - submariner: <commit-id> 
+      - lighthouse: <commit-id>
+      - submariner-operator: <commit-id>
+      - submariner-charts: <commit-id>
+      - shipyard: <commit-id>
+      - admiral: <commit-id>
+
+   ```
+
+## verification of the release yaml
+
+The verification step would:
+* Verify the format of the yaml file
+
+* Verify that passed CI runs exist for the commit-ids proposed for the components
+
+* Verify that submariner-operator dependencies to projects in go.mod are pointing
+  to the proposed submariner, lighthouse, and shipyard releases, because this affects
+  the behaviour of `subctl verify` to be up-to-date with the e2e tests of those repositories.
+  Otherwise propose a PR for anything missing and stop.
+
+* Verify that projects have Dockerfile.dapper pinned to the specific shipyard release,
+  otherwise send a PR to the projects which didn't pin the image yet and stop.
+
+* Verify that submariner/lighthouse are using the specific admiral commit ID, otherwise send
+  a PR to submariner/lighthouse importing the specific commit ID, and stop.
+
+* Run E2E for the combination of commit-ids with helm & operator (with lighthouse enabled)
+
+## merge of the release yaml
+
+* A script will run in Github actions, executing step by step the tagging and verification
+of the multiple components.
+
+It will retry builds if necessary.
+
+* Shipyard will be tagged for reference
+
+* The release notes will be added to the website, proposed as a PR.
+
+* Shipyard Dockerfile.dapper will be unpinned from all repositories in master via PR.
+
+## Handling of versions.go in subctl/operator
+
+Subctl and the operator have a set of version references about the other project they manage,
+this is handled by updating a versions.go in the repository. For this management strategy
+this will need to change in the following way:
+
+### "latest" reference
+
+versions.go will reference latest by default in the source code, but will also work 
+as a placeholder for anybody compiling subctl with:
+
+* `-X github.com/submariner-io/submariner-operator/pkg/versions.LighthouseVersion=latest`
+* `-X github.com/submariner-io/submariner-operator/pkg/versions.SubmarinerVersion=latest`
+* `-X github.com/submariner-io/submariner-operator/pkg/versions.OperatorVersion=latest`
+
+### references pinned at compilation
+
+#### For a master build
+
+For the [devel bleeding edge](https://github.com/submariner-io/submariner-operator/releases/tag/devel)
+release channel, every component will be pinned at compilation time to the last successfully
+published image of each component (race conditions apart we assume this is what we tested
+in E2E). 
```

Comment:
Is this version for reference only, or does it affect the image the subctl actually deploys?

Reply:
Yeah that would be the simplest